include/simdjson/generic/ondemand/json_iterator-inl.h

```diff
@@ -43,9 +43,19 @@ inline void json_iterator::rewind() noexcept {
 SIMDJSON_PUSH_DISABLE_WARNINGS
 SIMDJSON_DISABLE_STRICT_OVERFLOW_WARNING
 simdjson_warn_unused simdjson_really_inline error_code json_iterator::skip_child(depth_t parent_depth) noexcept {
+  /***
+   * WARNING:
```

Comment:
Time and time again I get burned by calling skip_child while pointing at a key!!! We must not do that!!! Let us leave a warning from hell.

cc @jkeiser

Reply:
Hopefully there are ways not to get burned here--I think it would be possible to set depth+1 while you are at the key and the : (we'd have to adjust some other things too, but I'm pretty sure at least at one point I had this written that way). Obviously not for this PR.